django/db/backends/mysql/operations.py

```diff
@@ -177,13 +177,45 @@ def sql_flush(self, style, tables, sequences, allow_cascade=False):
         # NB: The generated SQL below is specific to MySQL
         # 'TRUNCATE x;', 'TRUNCATE y;', 'TRUNCATE z;'... style SQL statements
         # to clear all tables of all data
+        schema_name = self.connection.settings_dict['NAME']
+
         if tables:
+            with self.connection.cursor() as cursor:
+                cursor.execute("""
+                SELECT table_name, table_rows, auto_increment
+                FROM information_schema.tables
+                WHERE table_schema = %s;
```

Comment:
You want to filter by `table_name IN %` and pass `tables` in params else this will flush all tables instead of the ones specified by `available_apps`.

Reply:
If `table_name not in tables` is True, the table will be skipped at https://github.com/django/django/pull/12634/files#diff-a7516854a0eab5c74e491a6dc8249c2dR194-R195. So I guess the problem won't happen.

But it might be better to filter in the query level. What do you think?

Reply:
I think it's better to filter out at the database level to avoid retrieving all table rows.

Reply:
I see! I'll filter in the database level.